algorithms/linfa-clustering/src/gaussian_mixture/algorithm.rs

```diff
@@ -746,4 +753,33 @@ mod tests {
             ThreadRng::default(),
         ));
     }
-}
+
+    #[test]
+    fn test_predict_proba() {
+        let mut rng = Xoshiro256Plus::seed_from_u64(42);
+        let centroids = array![[0.0, 1.0], [-10.0, 20.0], [-1.0, 10.0]];
+        let n_samples_per_cluster = 1000;
+        let dataset = DatasetBase::from(generate::blobs(n_samples_per_cluster, &centroids, &mut rng));
+        let n_clusters = centroids.len_of(Axis(0));
+        // total samples = (n_samples_per_cluster * n_clusters)
+        let total_samples = n_samples_per_cluster * n_clusters;
+    
+        // model fitting
+        let gmm = GaussianMixtureModel::params(n_clusters)
+            .with_rng(rng)
+            .fit(&dataset)
+            .expect("Failed to fit GMM");
+    
+        // getting probabilities
+        let proba = gmm.predict_proba(dataset.records());
+    
+        // checking output shape is correct or not
+        assert_eq!(proba.dim(), (total_samples, n_clusters));
+    
+        // checking for each sample, the sum of probabilities is 1
+        for sample in proba.outer_iter() {
```

Comment:
You can use [sum_axis(Axis(1))](https://docs.rs/ndarray/0.15.6/ndarray/struct.ArrayBase.html#method.sum_axis) and check against `Array::ones(n_samples)`